src/main/java/com/authservice/authservice/controller/AuthenticationController.java

```diff
@@ -44,11 +47,23 @@ public AuthenticationController(AuthenticationManager authenticationManager, Jwt
     public ResponseEntity<?> createAuthenticationToken(@RequestBody JWTRequest authenticationRequest) throws Exception {
         log.info("Entering AuthenticationController.createAuthenticationToken with Parameter authenticationRequest {}.", authenticationRequest.toString());
         authenticate(authenticationRequest.getUserName(), authenticationRequest.getPassword());
-        final User userDetails = userDetailsServiceImpl.loadUserDetails(authenticationRequest.getUserName());
-        final String token = jwtTokenUtil.generateToken(userDetails);
+        final UserDetails userDetails = userDetailsServiceImpl.loadUserByUsername(authenticationRequest.getUserName());
+        final String token = jwtTokenUtil.generateToken((User) userDetails);
         return ResponseEntity.ok(new JWTResponse(token));
     }
 
+    /**
+     * This function will validate the token and return the userId
+     *
+     * @return
+     */
+    @GetMapping("/validating-token")
+    public ResponseEntity<AuthenticatedResponse> validatingToken(Authentication auth) {
```

Comment:
if you take directly principal it does not work ? are you sure ?

`auth.getPrincipal` you are already doing it does not work with Principal directly ?

Reply:
It wouldn't work with Principal.